test/optimizer/stats_catalog_test.cpp

```diff
@@ -0,0 +1,220 @@
+                             const catalog::table_oid_t table_oid, std::vector<storage::TupleSlot> *index_results) {
+    const auto &oid_pri = statistics_oid_index_->GetProjectedRowInitializer();
+    const auto &oid_prm = statistics_oid_index_->GetKeyOidToOffsetMap();
+
+    const std::unique_ptr<byte[]> buffer_lo(common::AllocationUtil::AllocateAligned(oid_pri.ProjectedRowSize()));
+    const std::unique_ptr<byte[]> buffer_hi(common::AllocationUtil::AllocateAligned(oid_pri.ProjectedRowSize()));
+
+    auto *const pr_lo = oid_pri.InitializeRow(buffer_lo.get());
+    auto *const pr_hi = oid_pri.InitializeRow(buffer_hi.get());
+
+    // Write the attributes in the ProjectedRow
+    // Low key (class, INVALID_COLUMN_OID)
+    *(reinterpret_cast<catalog::table_oid_t *>(pr_lo->AccessForceNotNull(oid_prm.at(catalog::indexkeycol_oid_t(1))))) =
+        table_oid;
+    *(reinterpret_cast<catalog::col_oid_t *>(pr_lo->AccessForceNotNull(oid_prm.at(catalog::indexkeycol_oid_t(2))))) =
+        catalog::col_oid_t(0);
+
+    const auto next_oid = catalog::table_oid_t(!table_oid + 1);
+    // High key (class + 1, INVALID_COLUMN_OID)
+    *(reinterpret_cast<catalog::table_oid_t *>(pr_hi->AccessForceNotNull(oid_prm.at(catalog::indexkeycol_oid_t(1))))) =
+        next_oid;
+    *(reinterpret_cast<catalog::col_oid_t *>(pr_hi->AccessForceNotNull(oid_prm.at(catalog::indexkeycol_oid_t(2))))) =
+        catalog::col_oid_t(0);
+
+    statistics_oid_index_->ScanAscending(*txn, storage::index::ScanType::Closed, 2, pr_lo, pr_hi, 0, index_results);
+  }
+
+  std::unique_ptr<DBMain> db_main_;
+  common::ManagedPointer<transaction::TransactionManager> txn_manager_;
+  common::ManagedPointer<catalog::Catalog> catalog_;
+  catalog::db_oid_t db_;
+  common::ManagedPointer<storage::SqlTable> statistics_;
+  common::ManagedPointer<storage::index::Index> statistics_oid_index_;
+};
+
+/*
+ * Create and delete a user table.
+ */
+// NOLINTNEXTLINE
+TEST_F(StatsCatalogTests, StatisticTest) {
+  // Create a new table in the catalog, backed by a SqlTable
+  catalog::table_oid_t table_oid;
+  std::set<catalog::col_oid_t> col_oids;
+  {
+    auto txn = txn_manager_->BeginTransaction();
+    auto accessor = catalog_->GetAccessor(common::ManagedPointer(txn), db_);
+    EXPECT_NE(accessor, nullptr);
+
+    // Create the column definition (no OIDs)
+    std::vector<catalog::Schema::Column> cols;
+    cols.emplace_back("id", type::TypeId::INTEGER, false,
+                      parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    cols.emplace_back("user_col_1", type::TypeId::INTEGER, false,
+                      parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    auto tmp_schema = catalog::Schema(cols);
+
+    table_oid = accessor->CreateTable(accessor->GetDefaultNamespace(), "test_table", tmp_schema);
+    EXPECT_NE(table_oid, catalog::INVALID_TABLE_OID);
```

Comment:
Again this is a useless check. The table will always get created. Is this code copy-pasted?